### PR TITLE
Update various dependencies

### PR DESCRIPTION
**Pull request by shaopeng-gh**
Updating dependencies from bots, and fixed build script warning, there is no build script warning now.

### CODE REVIEW

File: BuildAndTest.cmd [hunk 1]
```diff
@@ -96,7 +96,7 @@ Exit /B %ERRORLEVEL%
 :CreatePublishPackage
 set Framework=%~1
 set RuntimeArg=%~2
-dotnet publish %~dp0src\BinSkim.Driver\BinSkim.Driver.csproj --no-restore -c %Configuration% -f %Framework% --runtime %RuntimeArg%
+dotnet publish %~dp0src\BinSkim.Driver\BinSkim.Driver.csproj --no-restore -c %Configuration% -f %Framework% --runtime %RuntimeArg% --self-contained
```
Comment (shaopeng-gh) [inline BuildAndTest.cmd:99]: >--self-contained [](http://example.com/codeflow?start=131&length=16)

fixing the warning btw,
asking us to specify if want self-contained or not. Default is self-contained, so no actual change.

File: ReleaseHistory.md [hunk 2]
```diff
@@ -16,10 +16,13 @@
 - NEW => new feature 
 
 ## UNRELEASED
+* DEP: Update `ELFSharp` package from 2.17.1 to 2.17.2. [#930](https://github.com/microsoft/binskim/pull/930)
+* DEP: Update `System.Reflection.Metadata` package from 7.0.0 to 7.0.2. [#930](https://github.com/microsoft/binskim/pull/930)
+* DEP: Update `Microsoft.CodeAnalysis.NetAnalyzers` package from 7.0.0 to 7.0.3. [#930](https://github.com/microsoft/binskim/pull/930)
```
Comment (shaopeng-gh) [inline ReleaseHistory.md:21]: >7.0.3 [](http://example.com/codeflow?start=74&length=5)

there was an an old note upgrading to 7.0.1 below,
I deleted that note.

File: src/Nuget/BinaryParsers.nuspec [hunk 3]
```diff
@@ -18,10 +18,10 @@
     <dependencies>
       <group targetFramework=".NETStandard2.1">
         <dependency id="Sarif.Driver" version="2.4.12" exclude="Build,Analyzers" />
-        <dependency id="ELFSharp" version="2.13.2" exclude="Build,Analyzers" />
```
Comment (shaopeng-gh) [inline src/Nuget/BinaryParsers.nuspec:21]: >ELFSharp [](http://example.com/codeflow?start=24&length=8)

This file was either missed, or on purpose have a specific old version stated here.
I think this file should be updated, let me know if wrong.

**Review by schlaman-ms — APPROVED** [review]
:shipit:

**Pull request merged**